src/main/java/net/dries007/tfc/objects/blocks/TFCBlocks.java

```diff
@@ -89,7 +93,14 @@
     private static <T extends Block> RegistryObject<T> register(String name, Supplier<T> blockSupplier, Item.Properties blockItemProperties)
     {
         RegistryObject<T> block = BLOCKS.register(name, blockSupplier);
-        TFCItems.ITEMS.register(name, () -> new BlockItem(block.get(), blockItemProperties));
+        TFCItems.ITEMS.register(name, () -> {
+            Block instance = block.get();
+            if (instance instanceof IBlockItemSupplier)
+            {
+                return ((IBlockItemSupplier) instance).apply(blockItemProperties);
+            }
+            return new BlockItem(block.get(), blockItemProperties);
+        });
```

Comment:
This really grinds my gears, but I can't really articulate why. The main reason is it spreads the registration (e.g. constructor properties) out into every single block, rather then everything being consolidated in one class (e.g. `TFCBlocks`). We do consolidate most item blocks rather than have separate `ItemBlock` registry objects in `TFCItems`, I'm not sure if that's good or bad. But this marker interface I really would rather avoid.

Reply:
Just this one, i'm not sure how else I can delegate the constructor for specific `Blockitem`s when needed besides using this method. If you have something in mind let me know